apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -518,10 +519,16 @@ type CPU struct {
 
 // Storage describes one storage device (disk, SSD, etc.) on the host.
 type Storage struct {
-	// The Linux device name of the disk, e.g. "/dev/sda". Note that this
-	// may not be stable across reboots.
+	// A Linux device name of the disk, e.g.
+	// "/dev/disk/by-path/pci-0000:01:00.0-scsi-0:2:0:0". This will be a name
+	// that is stable across reboots if one is available.
 	Name string `json:"name,omitempty"`
 
+	// A list of alternate Linux device names of the disk, e.g. "/dev/sda".
+	// Note that this list is not exhaustive, and names may not be stable
+	// across reboots.
+	AlternateNames []string `json:"alternateNames,omitempty"`
```

Comment:
To clarify this will be a list of known device names from Inspection **excluding** the one in `Name` field?

Also maybe we should `sort | uniq` this list somewhere?

Reply:
In the proposed implementation it _includes_ the name in the `name` field, and entries are unique. Also, the true kernel device (e.g. `/dev/sda`) always appears first.